### PR TITLE
Avereha/zipper ctx

### PR DESCRIPTION
**Pull request by avereha**
## What issue is this change attempting to solve?
zipper: 
move handler_test to http_handler_test to match the tested filename
group errors by message and count them

### CODE REVIEW

File: app/carbonzipper/http_handlers.go [hunk 1]
```diff
@@ -607,9 +609,10 @@ func errorsFanIn(ctx context.Context, errs []error, nBackends int) error {
 				"majority of backends returned not found. %d total errors, %d not found",
 				nErrs, nErrs-nNotNotFounds))
 		}
-
-		return errors.New(fmt.Sprintf(
-			"all backends failed with mixed errors: %d total errors, %d not found",
-			nErrs, nErrs-nNotNotFounds))
+		message := fmt.Sprintf("all backends failed with mixed errors: %+v", counts)
+		if len(message) > 300 {
```
Comment (grlvrl) [inline app/carbonzipper/http_handlers.go:613]: Why do we have a hard-coded "300" here?

Comment (avereha) [reply app/carbonzipper/http_handlers.go:613]: Was thinking to avoid huge logs.

Comment (grzkv) [reply app/carbonzipper/http_handlers.go:613]: I'd rather do that on a higher level to have this for all error messages at the same time, but that can be done later.

**Review by grzkv — APPROVED** [review]

File: app/carbonzipper/http_handlers.go (hunk 1 above)
Comment (avereha) [reply app/carbonzipper/http_handlers.go:613]: Filled https://github.com/bookingcom/carbonapi/issues/318 to continue this discussion.

**Pull request merged**